### ISSUE TITLE
Python error when packaging into DMG

### ISSUE THREAD

**Issue by Xam-Mlr** [electron-userland/electron-builder#6215]
* **22.11.7**: 

* **14.17.0**: 

* **13.2.1**:
* **Electron Type: current**:

* **DMG**: 

Hello, i've just finished my electron project and i'd like to make a DMG from my project.
I follow your do but i have an issue: when i enter "yarn dist", I have this error:

XXX@MacBookPro-2020 electron % yarn run dist
yarn run v1.22.11
$ electron-builder
  • electron-builder  version=22.11.7 os=20.5.0
  • loaded configuration  file=package.json ("build" field)
  • writing effective config  file=dist/builder-effective-config.yaml
  • packaging       platform=darwin arch=x64 electron=13.2.3 appOutDir=dist/mac
  • skipped macOS code signing  reason=identity explicitly is set to null
  • building        target=DMG arch=x64 file=dist/StegnHide-1.0.0.dmg
    ⨯ Exit code: 1. Command failed: /usr/bin/python /Users/XXX/Documents/MacBookPro/projects/electron/node_modules/dmg-builder/vendor/dmgbuild/core.py
  Traceback (most recent call last):
    File "/Users/XXX/Documents/MacBookPro/projects/electron/node_modules/dmg-builder/vendor/dmgbuild/core.py", line 31, in <module>
      from badge import badge
    File "/Users/XXX/Documents/MacBookPro/projects/electron/node_modules/dmg-builder/vendor/dmgbuild/badge.py", line 4, in <module>
      from Quartz import *
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/Quartz/__init__.py", line 6, in <module>
      import Foundation
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/Foundation/__init__.py", line 9, in <module>
      import CoreFoundation
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/CoreFoundation/__init__.py", line 22, in <module>
      import CoreFoundation._CoreFoundation
  RuntimeError: Wrong version of PyObjC C API (got 19, expected 20)
  
  Traceback (most recent call last):
    File "/Users/XXX/Documents/MacBookPro/projects/electron/node_modules/dmg-builder/vendor/dmgbuild/core.py", line 31, in <module>
      from badge import badge
    File "/Users/XXX/Documents/MacBookPro/projects/electron/node_modules/dmg-builder/vendor/dmgbuild/badge.py", line 4, in <module>
      from Quartz import *
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/Quartz/__init__.py", line 6, in <module>
      import Foundation
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/Foundation/__init__.py", line 9, in <module>
      import CoreFoundation
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/CoreFoundation/__init__.py", line 22, in <module>
      import CoreFoundation._CoreFoundation
  RuntimeError: Wrong version of PyObjC C API (got 19, expected 20)
    failedTask=build stackTrace=Error: Exit code: 1. Command failed: /usr/bin/python /Users/XXX/Documents/MacBookPro/projects/electron/node_modules/dmg-builder/vendor/dmgbuild/core.py
  Traceback (most recent call last):
    File "/Users/XXX/Documents/MacBookPro/projects/electron/node_modules/dmg-builder/vendor/dmgbuild/core.py", line 31, in <module>
      from badge import badge
    File "/Users/XXX/Documents/MacBookPro/projects/electron/node_modules/dmg-builder/vendor/dmgbuild/badge.py", line 4, in <module>
      from Quartz import *
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/Quartz/__init__.py", line 6, in <module>
      import Foundation
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/Foundation/__init__.py", line 9, in <module>
      import CoreFoundation
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/CoreFoundation/__init__.py", line 22, in <module>
      import CoreFoundation._CoreFoundation
  RuntimeError: Wrong version of PyObjC C API (got 19, expected 20)
  
  Traceback (most recent call last):
    File "/Users/XXX/Documents/MacBookPro/projects/electron/node_modules/dmg-builder/vendor/dmgbuild/core.py", line 31, in <module>
      from badge import badge
    File "/Users/XXX/Documents/MacBookPro/projects/electron/node_modules/dmg-builder/vendor/dmgbuild/badge.py", line 4, in <module>
      from Quartz import *
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/Quartz/__init__.py", line 6, in <module>
      import Foundation
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/Foundation/__init__.py", line 9, in <module>
      import CoreFoundation
    File "/Users/XXX/Library/Python/2.7/lib/python/site-packages/CoreFoundation/__init__.py", line 22, in <module>
      import CoreFoundation._CoreFoundation
  RuntimeError: Wrong version of PyObjC C API (got 19, expected 20)

    at /Users/XXX/Documents/MacBookPro/projects/electron/node_modules/builder-util/src/util.ts:132:18
    at ChildProcess.exithandler (child_process.js:326:5)
    at ChildProcess.emit (events.js:376:20)
    at maybeClose (internal/child_process.js:1055:16)
    at Socket.<anonymous> (internal/child_process.js:441:11)
    at Socket.emit (events.js:376:20)
    at Pipe.<anonymous> (net.js:673:12)
  error Command failed with exit code 1.
  info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

That's my package.json:

`{
  "name": "xxx",
  "version": "1.0.0",
  "description": "xxx",
  "main": "main.js",
  "scripts": {
    "postinstall": "install-app-deps",
    "start": "electron .",
    "build": "electron-builder . StegnHide",
    "dist": "electron-builder"
  },
  "author": "xxx",
  "license": "MIT",
  "build": {
    "mac": {
      "target": ["dmg"],
      "identity": null,
      "icon": "build/icon.icns",
      "category": "utilities"
    },
    "dmg": {
      "background": "build/background.tiff",
      "icon": "build/icon.icns",
      "iconSize": 150,
      "window": {
        "width": 800,
        "height": 600
      },
      "contents": [
        {
          "x": 85,
          "y": 250
        },
        {
          "x": 565,
          "y": 250,
          "type": "link",
          "path": "/Applicaions"
        }
      ]
    }
  },
  "devDependencies": {
    "electron": "^13.2.1",
    "electron-packager": "^15.3.0",
    "electron-builder": "^22.11.7"
  },
  "dependencies": {
    "python-shell": "^3.0.0"
  }
}
`

I have the same problem using npm

Thanks

**Comment by mmaietta** [comment]
Looks like you may need a different version of python installed? I see this in the error logs, but honestly am not sure what it means.
```
RuntimeError: Wrong version of PyObjC C API (got 19, expected 20)
```

**Comment by Xam-Mlr** [comment]
> Looks like you may need a different version of python installed? I see this in the error logs, but honestly am not sure what it means.
> 
> ```
> 
> RuntimeError: Wrong version of PyObjC C API (got 19, expected 20)
> 
> ```

I solve it by installing pyobjc: 
Pip install pyobjc

**Comment by Xam-Mlr** [comment]
> Looks like you may need a different version of python installed? I see this in the error logs, but honestly am not sure what it means.
> 
> ```
> 
> RuntimeError: Wrong version of PyObjC C API (got 19, expected 20)
> 
> ```

Have you ever used python shell because after the packaging it's not working anymore, I already had try with electron packager and it had worked but now it's not working with electron builder

**Comment by mmaietta** [comment]
I've never had to configure python for any of my build machines when using electron-builder, so I'm honestly not familiar with the error/configuration you're experiencing. 🙁

**Comment by greggman** [comment]
I've never needed one other but now I'm getting

```
Error: Exit code: ENOENT. spawn /usr/bin/python ENOENT
    at /Users/gregg/src/servez/node_modules/builder-util/src/util.ts:133:18
    at exithandler (node:child_process:404:5)
    at ChildProcess.errorhandler (node:child_process:416:5)
    at ChildProcess.emit (node:events:390:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
    at onErrorNT (node:internal/child_process:477:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
    at throwError (/Users/gregg/src/servez/node_modules/builder-util/src/asyncTaskManager.ts:88:11)
    at checkErrors (/Users/gregg/src/servez/node_modules/builder-util/src/asyncTaskManager.ts:53:9)
    at AsyncTaskManager.awaitTasks (/Users/gregg/src/servez/node_modules/builder-util/src/asyncTaskManager.ts:58:5)
    at Packager.doBuild (/Users/gregg/src/servez/node_modules/app-builder-lib/src/packager.ts:453:23)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Object.executeFinally (/Users/gregg/src/servez/node_modules/builder-util/src/promise.ts:12:14)
    at Packager._build (/Users/gregg/src/servez/node_modules/app-builder-lib/src/packager.ts:376:31)
    at Packager.build (/Users/gregg/src/servez/node_modules/app-builder-lib/src/packager.ts:337:12)
    at Object.executeFinally (/Users/gregg/src/servez/node_modules/builder-util/src/promise.ts:12:14)
```

which suggests python is a dependency not listed anywhere?

**Comment by greggman** [comment]
I never have either but on a new mac I'm getting this

Error: Exit code: ENOENT. spawn /usr/bin/python ENOENT
    at /Users/gregg/src/myproj/node_modules/builder-util/src/util.ts:133:18
    at exithandler (node:child_process:404:5)
    at ChildProcess.errorhandler (node:child_process:416:5)
    at ChildProcess.emit (node:events:390:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
    at onErrorNT (node:internal/child_process:477:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
    at throwError (/Users/gregg/src/myproj/node_modules/builder-util/src/asyncTaskManager.ts:88:11)
    at checkErrors (/Users/gregg/src/myproj/node_modules/builder-util/src/asyncTaskManager.ts:53:9)
    at AsyncTaskManager.awaitTasks (/Users/gregg/src/myproj/node_modules/builder-util/src/asyncTaskManager.ts:58:5)
    at Packager.doBuild (/Users/gregg/src/myproj/node_modules/app-builder-lib/src/packager.ts:453:23)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Object.executeFinally (/Users/gregg/src/myproj/node_modules/builder-util/src/promise.ts:12:14)
    at Packager._build (/Users/gregg/src/myproj/node_modules/app-builder-lib/src/packager.ts:376:31)
    at Packager.build (/Users/gregg/src/myproj/node_modules/app-builder-lib/src/packager.ts:337:12)
    at Object.executeFinally (/Users/gregg/src/myproj/node_modules/builder-util/src/promise.ts:12:14)
```

And I also see https://github.com/electron-userland/electron-builder/blob/2def112bc1ac42046b921206825871b82ebf0955/packages/dmg-builder/src/dmg.ts#L300

But there is no python on MacOS anymore

**Comment by danielweck** [comment]
Solution: https://github.com/electron-userland/electron-builder/issues/6764

**Comment by mmaietta** [comment]
Resolved in 23.0.3
